google/cloud/internal/filesystem.h

```diff
@@ -167,6 +167,9 @@ inline bool is_other(file_status s) noexcept {
 std::uintmax_t file_size(std::string const& path);
 std::uintmax_t file_size(std::string const& path, std::error_code& ec) noexcept;
 
+/// Append @p path (even if it is an absolute directory) to @p directory
```

Comment:
Nit: The directory-ness of path isn't a factor.

Reply:
Fixed.